src/Symfony/Bundle/FrameworkBundle/DataCollector/MessagesDataCollector.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DataCollector;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+use Symfony\Component\Message\MiddlewareInterface;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class MessagesDataCollector extends DataCollector implements MiddlewareInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function collect(Request $request, Response $response, \Exception $exception = null)
+    {
+        return $this->data;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'messages';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reset()
+    {
+        $this->data = array();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle($message, callable $next)
+    {
+        $debugRepresentation = array(
+            'message' => array(
+                'type' => get_class($message),
+            ),
+        );
+
+        try {
+            $result = $next($message);
+
+            if (is_object($result)) {
+                $debugRepresentation['result'] = array(
```

Comment:
what about `$debugRepresentation['result'] = $result` with pretty dump in the profiler.

Reply:
I'm afraid it could be a massive object and wouldn't bring benefits for now. Should we postpone such decision to when we have the real problem?